Add jinja2 to prerequisites in README-CMAKE.md

**Describe the bug**
I am using CMake to build the repository. Using the following command:

```shell
cmake -S . -B build -DCMAKE_INSTALL_PREFIX=/opt/homebrew/Cellar/wpilib/2023.4.3 -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_VERBOSE_MAKEFILE=ON -Wno-dev -DBUILD_TESTING=OFF -DCMAKE_OSX_SYSROOT=/Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk
```

I get this error:

```
Traceback (most recent call last):
  File "/tmp/wpilib-20231016-3394-ellcrb/allwpilib-2023.4.3/ntcore/generate_topics.py", line 4, in <module>
    from jinja2 import Environment, FileSystemLoader
ModuleNotFoundError: No module named 'jinja2'
CMake Error at ntcore/CMakeLists.txt:11 (message):
  python and python3 generate_topics.py failed
```

**To Reproduce**
Steps to reproduce the behavior:
1. Install `cmake` (v3.27.7)
2. Install `opencv` (v4.8.1)
3. Install `openjdk` (v21)
4. Run the CMake command shown above

**Expected behavior**
Expected to build all the libraries listed in `README-CMAKE.md`.

**Desktop**
 - OS: macOS 14.1 Beta (23B5067a)
 - Project Information: WPILib v2023.4.3, CMake v3.27.7, OpenCV v4.8.1, OpenJDK v21

**Additional context**
In case the context matters, I am creating a Homebrew formula for WPILib.

--- a/README-CMAKE.md
+++ b/README-CMAKE.md
@@ -18,9 +18,11 @@ By default, all libraries except for the HAL and WPILib get built with a default
 
 ## Prerequisites
 
-The most common prerequisite is going to be OpenCV. OpenCV needs to be findable by CMake. On systems like the Jetson, this is installed by default. Otherwise, you will need to build OpenCV from source and install it.
+The jinja2 pip package is needed to generate classes for NT4's pubsub.
 
-In addition, if you want JNI and Java, you will need a JDK of at least version 11 installed. In addition, you need a `JAVA_HOME` environment variable set properly and set to the JDK directory.
+OpenCV needs to be findable by CMake. On systems like the Jetson, this is installed by default. Otherwise, you will need to build OpenCV from source and install it.
+
+If you want JNI and Java, you will need a JDK of at least version 11 installed. In addition, you need a `JAVA_HOME` environment variable set properly and set to the JDK directory.
 
 If you are building with unit tests or simulation modules, you will also need an Internet connection for the initial setup process, as CMake will clone google-test and imgui from GitHub.
 

